org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacUtils.java

```diff
@@ -357,6 +357,10 @@ public static <T> boolean isEmpty(List<T> list) {
 	private static Collection<File> classpathEntriesToFiles(JavaProject project, Predicate<IClasspathEntry> select) {
 		try {
 			LinkedHashSet<File> res = new LinkedHashSet<>();
+			IPath outputLocation = project.getOutputLocation();
+			if (outputLocation != null) {
+				addPath(project, outputLocation, res);
+			}
```

Comment:
Similar operation is already performed on line 307 with `classpathFiles.addAll(outDirectories(javaProject, entry -> isTest || !entry.isTest()));`.

Reply:
This code is called only for a modular project. I have updated the PR.